ddl/table.go

```diff
@@ -72,7 +72,15 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		}
 		if atomic.LoadUint32(&EnableSplitTableRegion) != 0 {
 			// TODO: Add restrictions to this operation.
-			go splitTableRegion(d.store, tbInfo.ID)
+			pi := tbInfo.GetPartitionInfo()
+			if pi != nil {
+				// Max partition count is 4096, should we sample and just choose some of the partition to split?
```

Comment:
PTAL @disksing

Reply:
I think we can split it to 4096 regions. But it could be very slow if you do it in parallel.

Reply:
Why "it could be very slow", any suggestion here? @disksing

Reply:
It may take a very long time for tikv to complete splitting the region into 4096 regions.

Reply:
Could `AskBatchSplit` be faster? @disksing

Reply:
I think it can be faster but not sure if it is allowed to call `BatchSplit` from tidb (I only know PD can send batch split command). /cc @overvenus

Reply:
The current suggestion is to send requests one by one synchronously because if you send them all together, a successful split will cause others to report staleEpoch error. Later after we introduce batch split support, then you can swith to use batchSplit request.